executor/src/witgen/sequence_iterator.rs

```diff
@@ -59,7 +53,7 @@ impl DefaultSequenceIterator {
     /// If we're not at the last identity in the current row, just moves to the next.
     /// Otherwise, starts with identity 0 and moves to the next row if no progress was made.
     fn update_state(&mut self) {
-        while !self.is_done() && !self.has_more_actions() {
```

Comment:
This `while` was equivalent to an if: `has_more_actions()` computes `self.cur_action_index < last_action_index` where `last_action_index >= 0`, and `start_next_round()` sets `self.cur_action_index = -1;`